### PR TITLE
Adding table prefix handling

### PR DESCRIPTION
**Pull request by burzum**
This works as long as _all_ tables have the prefix in the selected DB.

### CODE REVIEW

File: src/Shell/Task/ModelTask.php [hunk 1]
```diff
@@ -85,6 +85,14 @@ public function main($name = null)
         parent::main();
         $name = $this->_getName($name);
 
+        if (!empty($this->connection)) {
+            $connection = ConnectionManager::get($this->connection);
+            $config = $connection->config();
+            if (isset($config['prefix'])) {
+                $this->tablePrefix = $config['prefix'];
+            }
+        }
```
Comment (lorenzo) [inline src/Shell/Task/ModelTask.php]: I would much rather provide the prefix via a command line option, instead of the connection config.

Comment (burzum) [inline src/Shell/Task/ModelTask.php]: I have a trait for the fixture and the table that use it based on the connection as well.

Comment (lorenzo) [inline src/Shell/Task/ModelTask.php]: Ok, but that is something bake should not care about. It is much cleaner and easier to test if you provide it yourself from the cli

**Comment by ionas** [comment]
command line prefix allows mix of non prefixed and prefixed tables?

**Comment by lorenzo** [comment]
@ionas it allows whatever meaning we convey to it

**Comment by ionas** [comment]
Great, the original statement at the top says otherwise, that's the reason I do ask so specifically.

**Comment by burzum** [comment]
@ionas well, @lorenzo wanted me to change it.

**Comment by ionas** [comment]
I hope you can work with it, chaining bake commands with && or ;; and --force should do magic.
Thanks for this contribution!

**Comment by markstory** [comment]
I feel like we should have at least one test related to this feature :smile:

File: src/Shell/Task/ModelTask.php [hunk 2]
```diff
@@ -299,6 +300,10 @@ public function findTableReferencedBy($schema, $keyField)
                 if (!isset($constraintInfo['references'])) {
                     continue;
                 }
+                $length = strlen($this->tablePrefix);
```
Comment (dereuromark) [inline src/Shell/Task/ModelTask.php]: I can already see people crying because they *must* use umlauts or special char prefixes `äöü` ^^.

Comment (burzum) [inline src/Shell/Task/ModelTask.php]: It's a valid point when you have to use Cake on a legacy DB with for example German table names.

Comment (dereuromark) [inline src/Shell/Task/ModelTask.php]: Yeah, better playing it safe using mb_ IMO.

Comment (burzum) [inline src/Shell/Task/ModelTask.php]: Agreed :+1:

**Comment by dereuromark** [comment]
Once the strlen() issue is fixed we can probably finalize and ship this maybe?

**Comment by lorenzo** [comment]
Should we close this?

**Comment by burzum** [comment]
Whats missing? Just the mb_strlen and a test?

**Comment by dereuromark** [comment]
And rebasing.

**Comment by burzum** [comment]
S

**Pull request closed without merging**